stages/org.osbuild.systemd

```diff
@@ -40,16 +53,75 @@ SCHEMA = """
   "default_target": {
     "type": "string",
     "description": "The default target to boot into"
+  },
+  "unit_dropins": {
+    "additionalProperties": false,
+    "type": "object",
+    "description": "Systemd unit drop-in configurations.",
+    "patternProperties": {
+      "^[\\w:.\\\\-]+[@]{0,1}[\\w:.\\\\-]*\\.service$": {
+        "additionalProperties": false,
+        "type": "object",
+        "description": "Drop-in configurations for a '.service' unit.",
+        "patternProperties": {
+          "^[\\w.-]{1,250}\\.conf$": {
+            "additionalProperties": false,
+            "type": "object",
+            "description": "Drop-in configuration for a '.service' unit.",
+            "properties": {
+              "service": {
+                "additionalProperties": false,
+                "type": "object",
+                "description": "'Service' configuration section of a unit file.",
+                "properties": {
+                  "environment": {
+                    "type": "string",
+                    "description": "Sets environment variables for executed process."
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
   }
 }
 """
 
 
+def configure_unit_dropins(tree, unit_dropins_options):
+    for unit, unit_dropins in unit_dropins_options.items():
+        # ensure the unit name + ".d" does not exceed maximum filename length
+        if len(unit+".d") > 255:
+            print(f"Error: the {unit} unit drop-in directory exceeds the maximum filename length.")
+            return 1
```

Comment:
Maybe just throw a `ValueError` exception, especially since that should not occur since the schema verification should catch it. In the mid-term I wanna wrap all the `main` functions in stages with [`api.exception_handler`](https://github.com/osbuild/osbuild/blob/d62c8294e1f9f7c6108a34b0202acd65b9b5fc67/osbuild/api.py#L235) which would send them to osbuild in a structured way.

Reply:
Well, the schema won't catch it, because I was not able to figure out how to limit the maximum size of the matched pattern, especially when there are multiple character sets which may have variable number of repetitions, but combined, they must be less than a specific size.

Raising the exception makes sense.

Reply:
Fixed